File: Python/Samples/Mediator/LoanColleague.py
from Colleague import Colleague
from Mediator import Mediator

class LoanColleague(Colleague):

    _mediator = None

    def __init__(self):
        self.prod = "房貸"

    @property
    def mediator(self):
        return self._mediator

    @mediator.setter
    def mediator(self, val=Mediator):
        self._mediator = val

    def score(self):
        # Implement the real score model here.
        return 30

File: CSharp/DP.Domain/Samples/Mediator/LoanColleague.cs
namespace DP.Domain.Samples.Mediator
{
    public class LoanColleague : IColleague
    {
        public string Prod { get; set; } = "房貸";
        public IMediator Mediator { get; set; }
        public LoanColleague(IMediator mediator=null)
        {
            this.Mediator = mediator;
        }
        public decimal Score()
        {
            //Implement the real score model here.
            return 30;
        }
    }
}